amlb/datasets/openml.py

```diff
@@ -122,13 +123,18 @@ def get_values(f):
 
         has_missing_values = lambda f: f.number_missing_values > 0
         is_target = lambda f: f.name == self._oml_task.target_name
-        return [Feature(f.index,
+        feat_indexer = iter(count())
+        return [Feature(new_idx,
                         f.name,
                         f.data_type,
                         values=get_values(f),
                         has_missing_values=has_missing_values(f),
                         is_target=is_target(f)
-                        ) for i, f in sorted(self._oml_dataset.features.items())]
+                        )
+                for new_idx, f in zip(feat_indexer,
```

Comment:
Any particular reason this is not just done with the enumerate built-in?

Reply:
code is a living thing :)
looks completely unnecessary now, was introduced with a previous (and wrong) approach.